File: app/src/main/java/com/suzei/racoon/ui/auth/forgotpassword/ForgotPasswordContract.java
package com.suzei.racoon.ui.auth.forgotpassword;

import com.suzei.racoon.ui.base.Contract;

public interface ForgotPasswordContract {

    interface View extends Contract.ProgressView {

        void setPresenter(Presenter presenter);

        void onResetPasswordSuccess();

        void onResetPasswordFailure(String message);

    }

    interface Presenter {

        void resetPassword(String email);

    }

}

File: app/src/main/java/com/suzei/racoon/ui/auth/forgotpassword/ForgotPasswordContract.kt
package com.suzei.racoon.ui.auth.forgotpassword

import com.suzei.racoon.ui.base.Contract.ProgressView

interface ForgotPasswordContract {

    interface View : ProgressView {
        fun setPresenter(presenter: Presenter)
        fun onResetPasswordSuccess()
        fun onResetPasswordFailure(message: String)
    }

    interface Presenter {
        fun resetPassword(email: String)
    }
}